src/api/store.api.ts

```diff
@@ -0,0 +1,19 @@
+import { firebaseCollection } from '@/model/collection.model';
+import { ShopStatus } from '@/model/shop.model';
+import { addToDb } from '@/services/api.service';
+import { RootState } from '@/state/store';
+import { useSelector } from 'react-redux';
+import { toast } from 'sonner';
+
+export default abstract class StoreApiService {
+  static addNewStore = async (values) => {
+    const storeInfo = {
+      ...values,
+      status: ShopStatus.pending,
+    };
+    toast.loading('adding your store...', { id: 'loading' });
+    await addToDb(storeInfo, firebaseCollection.Stores);
+    toast.dismiss('loading');
+    toast.success('store added successfully');
+  };
+}
```

Comment:
You may rename `src/api` folder to `src/services` and update the corresponding files to use `.service` naming convention. `api` keyword is generally associated with BE endpoints(or your custom endpoints) from FE pov.